### PR TITLE
qdl: init at unstable-2021-05-06

### PR DESCRIPTION
**Pull request by musfay**
###### Motivation for this change
A great tool to flash system images to Qualcomm devices.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/qdl/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  installPhase = ''
+    install -Dm755 ./qdl $out/bin/qdl
+  '';
+
+  meta = with lib; {
+    homepage    = "https://github.com/andersson/qdl";
+    description = "A tool to flash images to Qualcomm devices.";
+    license     = licenses.bsd3;
+    maintainers = with maintainers; [ ];
```
Comment (samuelgrf) [inline pkgs/tools/misc/qdl/default.nix:24]: Needs a maintainer.

File: pkgs/tools/misc/qdl/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner  = "andersson";
+    repo   = "qdl";
+    rev    = "2021b303a81ca1bcf21b7f1f23674b5c8747646f";
+    sha256 = "0akrdca4jjdkfdya36vy1y5vzimrc4pp5jm24rmlw8hbqxvj72ri";
+  };
+
+  buildInputs = [ systemd libxml2 ];
+
+  installPhase = ''
```
Comment (samuelgrf) [inline pkgs/tools/misc/qdl/default.nix:16]:
```suggestion
  installPhase = ''
    runHook preInstall
```

File: pkgs/tools/misc/qdl/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    install -Dm755 ./qdl $out/bin/qdl
```
Comment (samuelgrf) [inline pkgs/tools/misc/qdl/default.nix:17]:
```suggestion
    install -Dm755 ./qdl $out/bin/qdl
    runHook postInstall
```

File: pkgs/tools/misc/qdl/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenv, fetchFromGitHub, libxml2, systemd}:
+
+stdenv.mkDerivation {
+  pname   = "qdl";
+  version = "20210506";
```
Comment (samuelgrf) [inline pkgs/tools/misc/qdl/default.nix:5]:
```suggestion
  version = "unstable-2021-05-06";
```

**Review by samuelgrf — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125188` at 6462ae8e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>qdl</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/qdl/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/misc/qdl/default.nix:24:5:
```
   |
24 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

**Comment by musfay** [comment]
@samuelgrf Should I add myself to `maintainer-list.nix` with this PR or create a new one?

**Comment by samuelgrf** [comment]
@musfay In this PR.
It should be done in a seperate commit with the message `maintainers: add <name-here>`, also make sure that commit comes before the one adding the package.

File: pkgs/tools/misc/qdl/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 ./qdl $out/bin/qdl
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage    = "https://github.com/andersson/qdl";
+    description = "A tool to flash images to Qualcomm devices.";
```
Comment (samuelgrf) [inline pkgs/tools/misc/qdl/default.nix:24]:
```suggestion
    description = "Tool for flashing images to Qualcomm devices";
```
See https://nixos.org/manual/nixpkgs/unstable/#var-meta-description .

File: pkgs/tools/misc/qdl/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 ./qdl $out/bin/qdl
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage    = "https://github.com/andersson/qdl";
+    description = "Tool to flash images to Qualcomm devices";
```
Comment (samuelgrf) [inline pkgs/tools/misc/qdl/default.nix:24]:
```suggestion
    description = "Tool for flashing images to Qualcomm devices";
```
The grammar seems wrong.

**Comment by samuelgrf** [comment]
Result of `nixpkgs-review pr 125188` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qdl</li>
  </ul>
</details>

**Review by samuelgrf — APPROVED** [review]
LGTM 👍

File: pkgs/tools/misc/qdl/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    repo   = "qdl";
+    rev    = "2021b303a81ca1bcf21b7f1f23674b5c8747646f";
+    sha256 = "0akrdca4jjdkfdya36vy1y5vzimrc4pp5jm24rmlw8hbqxvj72ri";
+  };
+
+  buildInputs = [ systemd libxml2 ];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 ./qdl $out/bin/qdl
```
Comment (sikmir) [inline pkgs/tools/misc/qdl/default.nix:18]:
```suggestion
    install -Dm755 ./qdl -t $out/bin
```

**Pull request merged**